### DISCUSSION TITLE
SQL ORM issue with limit clause on informix

### DISCUSSION THREAD

**Discussion by joseajunior** [sqlalchemy/sqlalchemy#10663] [Usage Questions] [answered]
I don't even know if this discussion should be here, but I'll try here first.
I'm using SqlAlchemy 2.0 with Informix database using ibm_db_sa, I'm doing a select statement using the limit clause and it's showing an SQL syntax error.

While I was debugging this error using ` echo=True `, I noticed that the ` limit_clause ` is adding a ` FETCH FIRST n ROWS ONLY ` to the SQL statement but this doesn't work, It needs to add ` LIMIT n `.

Debugging i little more, I noticed that on ` sqlalchemy/sql/compiler.py ` in the method ` _row_limit_clause ` instead of going to the ` limit_clause ` of this class is going to the function under ` ibm_db_sa/base.py `.

So this is expected? Because the ` self ` in that moment is the ibm_db_sa class instead of the sqlalchemy class.

Let me know if I need to move this discussion to the ibm_db_sa repo.

**Comment by zzzeek** [comment, marked as answer]
hi -

you'd need to report this to ibm_db_sa.  that dialect is what would be responsible for ensuring special SQL clauses like TOP / LIMIT / FETCH etc. are rendered in the way that the particular backend database expects.

**Comment by joseajunior** [comment]
Perfect, thanks for the clarification.

**Comment by CaselIT** [comment]
you may also try using `.limit(...)` instead of `.fetch(...)` if you where using fetch

**Reply by joseajunior** [reply]
I'm using ` .limit(...) ` but the _ibm_db_sa_ dialect adds the ` FETCH FIRST n ROWS ONLY ` to the sql statement and for this database this fails.
